### PR TITLE
Drop support for Ruby 2.5 and earlier versions

### PR DESCRIPTION
**Pull request by yucao24hours**
I would like to let Seory work with later Ruby versions.

### CODE REVIEW

**Comment by yucao24hours** [comment]
@koic Could you please review this PR?

**Comment by koic** [comment]
Is this PR intended to drop Ruby 2.5 and lower? If so, can you specify `spec.required_ruby_version = '>= 2.6.0'` in the seory.gemspec?

**Comment by yucao24hours** [comment]
Yes, I'll drop support for Ruby 2.5 and earlier versions.

**Comment by yucao24hours** [comment]
I've changed PR title since this includes some changes not in .travis.yml but also gemspec.

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,15 @@
 language: ruby
 rvm:
-  - 1.9.3
-  - 2.0.0
-  - 2.1
+  - 2.6.6
+  - 2.7.2
```
Comment (koic) [inline .travis.yml:4]: Ruby 3.0 may be added to the CI matrix.

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+  - 3.0
```
Comment (koic) [inline .travis.yml:5]: Ah..., Ruby 3.0 matrix seems to fail. 

```console
Failures:
  1) Seory::Helper.rails_helper_methods defines shortcuts to rails helper method can call helper method
     Failure/Error: delegate(*args, {to: :helper})
     ArgumentError:
       Delegation needs a target. Supply a keyword argument 'to' (e.g. delegate :hello, to: :greeter).
     # /home/travis/.rvm/gems/ruby-3.0.0/gems/activesupport-6.1.3.1/lib/active_support/core_ext/module/delegation.rb:173:in `delegate'
     # ./lib/seory/helper.rb:7:in `rails_helper_methods'
     # ./spec/seory/helper_spec.rb:27:in `block (4 levels) in <top (required)>'
     # ./lib/seory/helper.rb:27:in `module_eval'
     # ./lib/seory/helper.rb:27:in `block in modulize'
     # ./lib/seory/helper.rb:24:in `initialize'
     # ./lib/seory/helper.rb:24:in `new'
     # ./lib/seory/helper.rb:24:in `modulize'
     # ./lib/seory/helper.rb:16:in `apply!'
     # ./spec/seory/helper_spec.rb:34:in `block (3 levels) in <top (required)>'
```

https://travis-ci.org/github/esminc/seory/jobs/765918053

I think Ruby 3.0 support can be opened by another PR. Can you revert this Ruby 3.0 matrix?

**Comment by koic** [comment]
JFYI, the Ruby 3.0 build passes below. Anyway, let's support Ruby 3.0 in a separate PR.

```diff
% git diff .
diff --git a/lib/seory/helper.rb b/lib/seory/helper.rb
index 81ccf8c..d44f20d 100644
--- a/lib/seory/helper.rb
+++ b/lib/seory/helper.rb
@@ -4,7 +4,7 @@ module Seory
   class Helper
     module Dsl
       def rails_helper_methods(*args)
-        delegate(*args, {to: :helper})
+        delegate(*args, to: :helper)
       end
     end

@@ -29,4 +29,3 @@ module Seory
     end
   end
 end
```

File: .travis.yml (hunk 2 above)
Comment (yucao24hours) [reply .travis.yml:5]: >I think Ruby 3.0 support can be opened by another PR.

I was just thinking the same thing. :slightly_smiling_face: I'll revert that changes.

**Comment by koic** [comment]
Thanks!

**Pull request merged**